manager/secret_manager.go

```diff
@@ -21,14 +27,30 @@ func (r *ResourceManager) CreateSecret(namespace string, secret *model.Secret) (
 func (r *ResourceManager) SecretExists(namespace string, name string) (exists bool, err error) {
 	var foundSecret *model.Secret
 	var statusError *errors.StatusError
+
 	foundSecret, err = r.kubeClient.SecretExists(namespace, name)
 	if err != nil {
 		if goerrors.As(err, &statusError) {
 			if statusError.ErrStatus.Reason == "NotFound" {
+				logging.Logger.Log.WithFields(log.Fields{
```

Comment:
@aleksandrmelnikov I don't think we need to log status error like separately, I think logging just the top most error would be sufficient.